arroyo-console/src/routes/connections/JsonForm.tsx

```diff
@@ -56,7 +57,16 @@ function StringWidget({
   errors: any;
   onChange: (e: React.ChangeEvent<any>) => void;
   readonly?: boolean;
+  resetField?: (field: string) => any;
 }) {
+  const onChangeWrapper = (e: React.ChangeEvent<any>) => {
+    onChange(e);
+
+    if (resetField && e.target.value == '') {
```

Comment:
Should this be before the parent onchange? For the connection page, we rely on onchange to synchronize the current state for validation

Reply:
The issue that since we want to set the value to undefined, we need to do that after onChange sets it to the empty string.